### ISSUE TITLE
It is difficult to scroll Grafana dashboards on iPad

### ISSUE THREAD

**Issue by pstibrany** [grafana/grafana#25360]
When viewing Grafana dashboard on iPad, one needs to be very careful where scroll action is initiated. This makes it difficult to use Grafana on iPad efficiently.

Initiating scrolling is only safe on areas highlighted with yellow color.

If you try to scroll with the finger on the panel, panel starts moving instead:

![IMG_404AE37D3C55-1](https://user-images.githubusercontent.com/895919/83748781-ff647180-a662-11ea-9101-64542604cb11.jpeg)

If you scroll down, there is even less area available for scrolling. Trying to scroll on left sidebar moves entire window instead (red arrows):

![IMG_9739BA62589B-1](https://user-images.githubusercontent.com/895919/83748762-fa9fbd80-a662-11ea-9023-aa2b2bbd333f.jpeg)

**What you expected to happen**:
I would like to be able to scroll with my finger anywhere on the page. I *never* want to move the panel.

**How to reproduce it (as minimally and precisely as possible)**: Use iPad and dashboard that doesn't fit one page.

**Environment**:
- Grafana version: Grafana v7.1.0-099791b7pre (e65dbcfea1)
- User OS & Browser: iPad OS 13.4.1, using built-in Safari browser

**Comment by Estrax** [comment]
That's basically a problem with scroll on all devices with touch screens - what gets classified as a scroll action and what is a click with hold and move. Maybe we should add an ability to lock the panels order, so the scroll is going to work properly on mobile?
cc @lukasztyrala @sarlinska @jessover9000

**Comment by pstibrany** [comment]
Would it be possible to distinguish between scrolling and drag&dropping gestures? On iPad, drag&drop needs to be initiated first.

**Comment by peterholmberg** [comment]
We disable draggable if on small devices as stated here

https://github.com/grafana/grafana/blob/03f241a67441df1e2552df088e0a3b14d26baa67/public/app/features/dashboard/dashgrid/DashboardGrid.tsx#L68

This was just a quick fix however, need to revisit this.

**Comment by jessover9000** [comment]
I've heard about this problem from external users on Android devices (not only tablets but also phones) as well. The usual behaviour for touch devices afaik would be that drag and drop only gets enabled after touching and holding an element for a second or so (e.g. rearranging apps on homescreen works that way). not sure how that works technically but could we do it that way as well?

**Comment by zoltanbedi** [comment]
related to #24892

**Comment by Pwuts** [comment]
@zoltanbedi yep, this is the same issue; happens if the viewport width >= 453px.

**Comment by alexhk** [comment]
Would it be a solution to 

- Dashboard Settings > General: If Editable is enabled, show a second option "Editable on Mobile Device",  
- Instead of using the resolution (current quick fix), check for mobile browser user agents 
(like in public/app/core/utils/browser.test.ts),

and let the user decide if they want it or not? 

It won't be 100% bullet proof with so many devices/resolutions, but it should should cover a large percentage of devices and use cases at least.

I found a couple of lists with the most popular mobile browsers which can be used for keywords, or Github has recent code (example: https://github.com/kaimallea/isMobile). Bonus points if mobile phone/tablet is distinguished and can be separately enabled/disabled.

Or alternatively as a quick work around, add draggable=0 to the URL to disable the chaos.

**Comment by benganellison** [comment]
just disable the move on touch as a default and have an edit button that can be enabled. and also allow for reordering using arrows like in home-assistant. a bit cumbersome but works. scrolling does not work in grafana at all at the moment which is worse.

example from home-assistant
![Screenshot 2021-01-12 at 08 35 28](https://user-images.githubusercontent.com/23585524/104283859-77c0ca00-54b1-11eb-948c-419af3152fae.png)

but move on touch does not seem to be the only problem. 
trying to scroll on any type of visualization just does not give the expected result. the screen is just still and the selection of data is not changing either. i would had expected both. scroll horizontally and i would expect cross hair change. scroll vertically and i would expect a scrolling behaviour.

**Comment by benganellison** [comment]
Just updated to latest beta (v7.5.0-12553pre) and now the experience is a lot better, I would say fixed.

**Comment by matschaffer** [comment]
7.5.1 seems to work much nicer on iphone as well.

**Comment by zoltanbedi** [comment]
Thanks for confirming this @matschaffer. I'm going to close this issue then. Let us know if it is still an issue and we can reopen it.

**Comment by pstibrany** [comment]
> Thanks for confirming this @matschaffer. I'm going to close this issue then. Let us know if it is still an issue and we can reopen it.

What was the fix? I missed that.
